src/server/database/Database/MySQLConnection.cpp

```diff
@@ -227,7 +227,11 @@ bool MySQLConnection::Execute(PreparedStatementBase* stmt)
 
     uint32 _s = getMSTime();
 
+#ifdef __APPLE__
```

Comment:
no need for this. just use mysql_stmt_bind_named_param().

mysql_stmt_bind_param() is deprecated and will be removed in future versions of MySQL

Reply:
The reason for the pull request was not to replace the method, but to solve the test problem in macOS. In Windows, I do not have access to that method, and if I change it, then I am obliged to update the library, which I will do at some point, but I prefer to leave it that way for now. Once we are sure that we cannot support previous versions (which still work), we can make that change that you say. But for now, the goal is to get it to pass testing on macOS only, not to upgrade MySQL.

Reply:
For the moment, I prefer to leave that condition and solve only the testing problem. I think that later, as you say, we can have that method used directly, but at least I, for example in Windows, do not have that method. I'm not sure what version of MySQL I'm using, but the one I have at the moment gives me an error, it says it can't find it. I think it is 8.0 and that method was added later. Better let's do things calmly, and in this particular case, let's fix the compilation problem on macOS, then we check the rest.

Reply:
I think mysql_stmt_bind_named_param is currently in 8.3, but in 8.0, for example, it is not. What happens to operating systems or distributions that cannot access that function? We cannot leave them without support. I prefer, therefore, to leave it currently as it is, and then when things become a little more standardized, then if I make the definitive change. Unless you, who are more experienced than me, have more information or think I'm wrong. Please, if so, let me know, and please provide me with information so I can read about it. Thank you. @Kitzunu

Reply:
Okay but what if someone builds Mac with MySQL 8.0.

Add a MySQL version check instead of a mac check :)

Reply:
> Okay but what if someone builds Mac with MySQL 8.0.
> 
> Add a MySQL version check instead of a mac check :)

I like this alternative better.

Reply:
I'm not sure, but I think that this method is added in 8.3, so if you have 8.3 or higher, use the new one and if not, use the old one.

Reply:
yea so something like

#if MYSQL_VER >= 803000u

or whatever the correct variable would be

Reply:
Thanks.